google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -2778,6 +2780,150 @@ public static Builder newBuilder() {
     }
   }
 
+  /**
+   * A class to contain all information needed for a Google Cloud Storage Object Move.
+   *
+   * @since 2.48.0
+   * @see Storage#moveBlob(MoveBlobRequest)
+   */
+  @TransportCompatibility({Transport.HTTP, Transport.GRPC})
+  final class MoveBlobRequest {
+    private final BlobId source;
+    private final BlobId target;
+    private final ImmutableList<BlobSourceOption> sourceOptions;
+    private final ImmutableList<BlobTargetOption> targetOptions;
+
+    MoveBlobRequest(
+        BlobId source,
+        BlobId target,
+        ImmutableList<BlobSourceOption> sourceOptions,
+        ImmutableList<BlobTargetOption> targetOptions) {
+      this.source = source;
+      this.target = target;
+      this.sourceOptions = sourceOptions;
+      this.targetOptions = targetOptions;
+    }
+
+    public BlobId getSource() {
+      return source;
+    }
+
+    public BlobId getTarget() {
+      return target;
+    }
+
+    public List<BlobSourceOption> getSourceOptions() {
+      return sourceOptions;
+    }
+
+    public List<BlobTargetOption> getTargetOptions() {
+      return targetOptions;
+    }
+
+    public Builder toBuilder() {
+      return new Builder(source, target, sourceOptions, targetOptions);
+    }
+
+    public static Builder newBuilder() {
```

Comment:
Do we maybe want to add something like MoveBlobRequest.of(BlobIdSource, BlobId target) for convenience?

Reply:
We could, but I sort of like having the builder methods that nudge people to at least think about preconditions. A move is only retryable if the target has a precondition set on it.